lib/cosmos/gui/qt.rb

```diff
@@ -531,14 +539,12 @@ def selection_end_line
   end
 
   private
-
   def escape_text(text, color)
+    @@color_cache[color] ||= "#%02X%02X%02X" % [color.red, color.green, color.blue]
     # You might think gsub! would use less memory but benchmarking proves gsub
     # with a regular express argument is the fastest and uses the least memory.
     # However, this is still an expensive operation due to how many times it is called.
-    text = text.gsub(/&/,'&amp;'.freeze).gsub(/\n/,'<br/>'.freeze).gsub(/\s/, '&nbsp;'.freeze).gsub(/>/,'&gt;'.freeze).gsub(/</,'&lt;'.freeze)
```

Comment:
Freeze saves memory but it's a method we were having to call each time and it was taking 5% of the processing time.